sale_product_set/models/product_set.py

```diff
@@ -1,11 +1,18 @@
 # -*- encoding: utf-8 -*-
-from openerp import fields, models, _
+from openerp import api, fields, models, _
 
 
 class ProductSet(models.Model):
     _name = 'product.set'
     _description = 'Product set'
 
-    name = fields.Char(_('Name'), help=_('Product set name'), required=True)
+    name = fields.Char(u"Name", help=u"Product set name", required=True)
     set_line_ids = fields.One2many(
-        'product.set.line', 'product_set_id', _(u"Products"))
+        'product.set.line', 'product_set_id', u"Products", copy=True)
+
+    @api.one
```

Comment:
``api.one`` is deprecated on v9. Assuming there is a single incoming record has always been a source of bugs, because you never know if some downstream code will do the same assumption. Maybe it'd a good idea to avoid it and put a loop instead?